### PR TITLE
pickup: add guns to holsters if picked up while unarmed

### LINKED ISSUE #1291: Add guns to emtpy holsters on pickup
This could be a small extra touch: In Natla's Mines (and likely TRLEs that strip you of pistols), picking them up will not add them to Lara's holsters. They remain empty until equipped once.

https://github.com/LostArtefacts/TR1X/assets/4934209/2b97254e-4df3-4e55-b6cd-19b07b5b5304

(sorry for the bad video crop)

Maybe they could appear in the holsters immediately after the pickup?

### PR DESCRIPTION
**Pull request by walkawayy**
Resolves #1291.

#### Checklist

- [X] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [X] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description
Add guns to Lara's empty holsters on pickup.

### CODE REVIEW

**Review by rr- — APPROVED** [review]

**Review by lahm86 — APPROVED** [review]

**Comment by aredfan** [comment]
@walkawayy Would it be possible to extend this fix to cover magnums and uzis as-well, assuming the holsters are empty?

There's a couple of scenarios in Natla's Mines applicable to this:

1. Instead of reclaiming the pistols, go straight to room 99 and collect the shotgun pickup.
2a. Then return to Cowboy and kill him with the shotgun to obtain the magnums.
2b. Or move forward and kill Skater Boy with the shotgun to obtain the uzis.

Screenshot

**Comment by walkawayy** [comment]
This PR now includes some renames to better clarify the object names to more clearly explain what's going on. The review is best looked at on a commit by commit basis. The majority of changes are just clarifications to object types and enforcing `GAME_OBJECT_ID` in some of the inventory functions I was looking at (ex- `item_num` was being used when it was actually an object). This should be clearer all around. I tested in Natla's Mines picking up each weapon while unarmed and picking up different combinations of weapons such as shotgun->pistols->uzis, uzis->pistols, etc.

File: src/game/objects/general/pickup.c [hunk 1]
```diff
@@ -49,16 +49,44 @@ static const OBJECT_BOUNDS m_PickUpBoundsUW = {
...
 };
 
+static void PickUp_AddGunToMesh(
+    const GAME_OBJECT_ID obj_num, ITEM_INFO *lara_item);
 static void PickUp_GetItem(
     int16_t item_num, ITEM_INFO *item, ITEM_INFO *lara_item);
 static void PickUp_GetAllAtLaraPos(ITEM_INFO *item, ITEM_INFO *lara_item);
 
+static void PickUp_AddGunToMesh(
+    const GAME_OBJECT_ID obj_num, ITEM_INFO *lara_item)
```
Comment (rr-) [inline src/game/objects/general/pickup.c]: `const ITEM_INFO *const lara_item`

File: src/game/objects/general/pickup.c [hunk 2]
```diff
@@ -49,16 +49,44 @@ static const OBJECT_BOUNDS m_PickUpBoundsUW = {
...
+{
+    bool lara_has_pistols = Inv_RequestItem(O_PISTOL_ITEM)
```
Comment (rr-) [inline src/game/objects/general/pickup.c]: const

File: src/game/objects/general/pickup.c [hunk 3]
```diff
@@ -49,16 +49,44 @@ static const OBJECT_BOUNDS m_PickUpBoundsUW = {
     },
 };
 
+static void PickUp_AddGunToMesh(
+    const GAME_OBJECT_ID obj_num, ITEM_INFO *lara_item);
```
Comment (rr-) [inline src/game/objects/general/pickup.c]: `const ITEM_INFO *lara_item`

File: src/game/inventory/inventory_func.c [hunk 4]
```diff
@@ -255,9 +258,9 @@ void Inv_RemoveAllItems(void)
     g_InvKeysCurrent = 0;
 }
 
-bool Inv_RemoveItem(int32_t item_num)
+bool Inv_RemoveItem(const GAME_OBJECT_ID object_id)
 {
-    int32_t item_num_option = Inv_GetItemOption(item_num);
+    const GAME_OBJECT_ID item_num_option = Inv_GetItemOption(object_id);
```
Comment (rr-) [inline src/game/inventory/inventory_func.c:264]: `inv_object_id`?

File: src/game/inventory/inventory_func.c [hunk 5]
```diff
@@ -227,18 +230,18 @@ void Inv_InsertItem(INVENTORY_ITEM *inv_item)
     }
 }
 
-int32_t Inv_RequestItem(int item_num)
+int32_t Inv_RequestItem(const GAME_OBJECT_ID object_id)
 {
-    int32_t item_num_option = Inv_GetItemOption(item_num);
+    const GAME_OBJECT_ID inv_object_option = Inv_GetItemOption(object_id);
```
Comment (rr-) [inline src/game/inventory/inventory_func.c]: `inv_object_id`

File: src/game/inventory/inventory_func.c [hunk 6]
```diff
@@ -9,32 +9,32 @@
 #include <stdbool.h>
 #include <stdint.h>
 
-bool Inv_AddItem(int32_t item_num)
+bool Inv_AddItem(const GAME_OBJECT_ID object_id)
 {
-    int32_t item_num_option = Inv_GetItemOption(item_num);
-    if (!g_Objects[item_num_option].loaded) {
+    const GAME_OBJECT_ID inv_object_option = Inv_GetItemOption(object_id);
```
Comment (rr-) [inline src/game/inventory/inventory_func.c]: `inv_object_id`

File: src/game/objects/general/pickup.c (hunk 1 above)
Comment (walkawayy) [reply src/game/objects/general/pickup.c:59]: So do we use full `const` usage if it's a static function declaration?

**Review by rr- — APPROVED** [review]

**Pull request merged**